datacube/drivers/postgres/_connections.py

```diff
@@ -123,7 +123,7 @@ def _create_engine(url, application_name=None, pool_timeout=60):
 
     @property
     def url(self) -> str:
-        return self._engine.url
+        return self._engine.url.create()
```

Comment:
This only works with the newest version of sqlalchemy, so will require changes to `setup.py` as well to constrain sqlalchemy version, alternative is to check sqlalchemy version or check whether `.create` method exists and fallback to previous behaviour otherwise

Reply:
Oh, right.

I reckon we should change `setup.py`. No looking back! What do you think?

Reply:
it will probably break a bunch of docker images, certainly test docker image will need to be updated.

Not too keen to force dependency on a very recent version of an important library that other packages depend on as well, possibly creating dependency hell.

```suggestion
        if hasattr(self._engin.url, 'create'):
            return self._engine.url.create()
        return str(self._engine.url)
```